graylog2-web-interface/src/components/search/SurroundingSearchButton.jsx

```diff
@@ -0,0 +1,65 @@
+import React from 'react';
+import {DropdownButton, MenuItem} from 'react-bootstrap';
+import naturalSort from 'javascript-natural-sort';
+import SearchStore from 'stores/search/SearchStore';
+
+import moment from 'moment';
+
+const SurroundingSearchButton = React.createClass({
+  propTypes: {
+    id: React.PropTypes.string.isRequired,
+    timestamp: React.PropTypes.number.isRequired,
+    searchConfig: React.PropTypes.object.isRequired,
+    messageFields: React.PropTypes.object.isRequired,
+  },
+
+  _buildTimeRangeOptions(searchConfig) {
+    const options = {};
+
+    Object.keys(searchConfig.surrounding_timerange_options).forEach((key) => {
+      options[moment.duration(key).asSeconds()] = searchConfig.surrounding_timerange_options[key];
+    });
+
+    return options;
+  },
+
+  _buildFilterFields() {
+    const fields = {};
+
+    if (this.props.searchConfig) {
+      this.props.searchConfig.surrounding_filter_fields.forEach((field) => fields[field] = this.props.messageFields[field]);
+    }
+
+    return fields;
+  },
+
+  _onClick(range) {
+    return (e) => {
+      e.preventDefault();
+
+      const fromTime = moment.unix(this.props.timestamp - parseInt(range, 10)).toISOString();
```

Comment:
What do we want to achieve with `parseInt`? I think we only use to convert the string `range` into a number, but then we should probably use `Number()` because of this:

> “parseInt() is useful for parsing strings, but you have to be aware that it stops at the first illegal digit. Parsing strings via Number() is less forgiving, but may produce nonintegers.”

> Excerpt From: Axel Rauschmayer. “Speaking JavaScript.”